crates/router/src/connector/datatrans/transformers.rs

```diff
@@ -1,235 +1,519 @@
+#[serde(rename_all = "camelCase")]
+pub struct PlainCardDetails {
+    #[serde(rename = "type")]
+    pub res_type: String,
+    pub number: cards::CardNumber,
+    pub expiry_month: Secret<String>,
+    pub expiry_year: Secret<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Default, Deserialize)]
+pub struct DatatransError {
+    pub code: String,
+    pub message: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum DatatransResponse {
+    TransactionResponse(DatatransSuccessResponse),
+    ErrorResponse(DatatransError),
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DatatransSuccessResponse {
+    pub transaction_id: String,
+    pub acquirer_authorization_code: String,
+}
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum DatatransRefundsResponse {
+    Success(DatatransSuccessResponse),
+    Error(DatatransError),
+}
+
+#[derive(Default, Debug, Serialize)]
+pub struct DatatransRefundRequest {
+    pub amount: MinorUnit,
+    pub currency: enums::Currency,
+    pub refno: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct DataPaymentCaptureRequest {
+    pub amount: MinorUnit,
+    pub currency: enums::Currency,
+    pub refno: String,
+}
+
+impl<T> TryFrom<(&api::CurrencyUnit, enums::Currency, MinorUnit, T)> for DatatransRouterData<T> {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        (_currency_unit, _currency, amount, item): (&api::CurrencyUnit, enums::Currency, i64, T),
+        (_currency_unit, _currency, amount, item): (
+            &api::CurrencyUnit,
+            enums::Currency,
+            MinorUnit,
+            T,
+        ),
     ) -> Result<Self, Self::Error> {
-        //Todo :  use utils to convert the amount to the type of amount that a connector accepts
         Ok(Self {
             amount,
             router_data: item,
         })
     }
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
-pub struct DatatransPaymentsRequest {
-    amount: i64,
-    card: DatatransCard,
-}
-
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
-pub struct DatatransCard {
-    number: cards::CardNumber,
-    expiry_month: Secret<String>,
-    expiry_year: Secret<String>,
-    cvc: Secret<String>,
-    complete: bool,
-}
-
 impl TryFrom<&DatatransRouterData<&types::PaymentsAuthorizeRouterData>>
     for DatatransPaymentsRequest
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
         item: &DatatransRouterData<&types::PaymentsAuthorizeRouterData>,
     ) -> Result<Self, Self::Error> {
+        if item.router_data.is_three_ds() {
+            return Err(errors::ConnectorError::NotImplemented(
+                "Three_ds payments through Datatrans".to_string(),
+            )
+            .into());
+        };
```

Comment:
Attach the example request/response in the PR desc that triggers this error, subsequently check the error message is required

Reply:
Attached